protocol/codec.go

```diff
@@ -246,6 +251,35 @@ func NewDecoderBytes(b []byte) Decoder {
 	return codec.NewDecoderBytes(b, CodecHandle)
 }
 
+// NewMsgpDecoderBytes returns a decoder object reading bytes from [b].
+// that works with msgp-serialized objects
+func NewMsgpDecoderBytes(b []byte) MsgpDecoder {
+	return &MsgpDecoderBytes{b: b, pos: 0}
+}
+
+// MsgpDecoderBytes is a []byte decoder into msgp-encoded objects
+type MsgpDecoderBytes struct {
+	b   []byte
+	pos int
+}
```

Comment:
We can avoid exporting `MsgpDecoderBytes`.
```suggestion
// NewMsgpDecoderBytes returns a decoder object reading bytes from [b].
// that works with msgp-serialized objects
func NewMsgpDecoderBytes(b []byte) *msgpDecoerBytes {
	return &msgpDecoderBytes{b: b, pos: 0}
}

// msgpDecoderBytes is a []byte decoder into msgp-encoded objects
type msgpDecoderBytes struct {
	b   []byte
	pos int
}
```

Reply:
brr
```
protocol/codec.go:251:36: exported func NewMsgpDecoderBytes returns unexported type *protocol.msgpDecoderBytes, which can be annoying to use
```